runtime/common/src/main/java/org/apache/nemo/runtime/common/metric/TaskMetric.java

```diff
@@ -106,6 +110,33 @@ private void setTaskDuration(final long taskDuration) {
     this.taskDuration = taskDuration;
   }
 
+  /**
+   * Method related to stream metric.
+   */
+  public final Map<String, List<StreamMetric>> getStreamMetric() {
+    return this.streamMetrics;
+  }
+
+  private void setStreamMetric(final Map<String, StreamMetric> streamMetricMap) {
+    for (String sourceVertexId : streamMetricMap.keySet()) {
+      StreamMetric streamMetric = streamMetricMap.get(sourceVertexId);
+      this.streamMetrics.putIfAbsent(sourceVertexId, new ArrayList<>());
+      this.streamMetrics.get(sourceVertexId).add(streamMetric);
+    }
+  }
+
+  /**
+   * Method related to latency.
+   */
+  public final Map<String, List<LatencyMetric>> getLatencymarks() {
+    return this.latencymarks;
+  }
+
+  private void addLatencymark(final LatencyMetric latencyMetric) {
+    this.latencymarks.putIfAbsent(latencyMetric.getLatencymark().getLastTaskId(), new ArrayList<>());
```

Comment:
Why `ArrayList`? Not `LinkedList`?

Reply:
I had a habit of using `ArrayList`. I'll change to `LinkedList`

Reply:
done